File: src/main/java/LDS/Person/service/WikiReviewService.java
package LDS.Person.service;

import LDS.Person.dto.request.WikiReviewCreateRequest;
import LDS.Person.dto.request.WikiReviewPageQueryRequest;
import LDS.Person.dto.request.WikiReviewUpdateRequest;
import LDS.Person.dto.response.PageResponse;
import LDS.Person.dto.response.WikiReviewListResponse;
import LDS.Person.dto.response.WikiReviewResponse;

/**
 * Wiki 审核服务接口
 */
public interface WikiReviewService {

    /**
     * 创建审核记录
     *
     * @param request 审核创建请求
     * @return 审核响应
     */
    WikiReviewResponse createReview(WikiReviewCreateRequest request);

    /**
     * 更新审核状态（1=通过并覆盖 Wiki 数据，2=拒绝）
     *
     * @param request 审核状态更新请求
     * @return 是否更新成功
     */
    boolean updateReviewStatus(WikiReviewUpdateRequest request);

    /**
     * 分页查询审核记录
     *
     * @param request 分页查询请求
     * @return 分页响应
     */
    PageResponse<WikiReviewListResponse> pageQuery(WikiReviewPageQueryRequest request);

    /**
     * 根据审核 ID 获取审核详情
     *
     * @param wikireviewId 审核记录 ID
     * @return 审核详情响应
     */
    WikiReviewResponse getReviewDetail(Long wikireviewId);
}

File: src/Services/WikiReviewService.cs
using SidusX.DTOs.Request;
using SidusX.DTOs.Response;
using SidusX.Models;
using SidusX.Repositories;

namespace SidusX.Services;

public class WikiReviewService
{
    private readonly WikiReviewRepository _repo;
    private readonly WikiRepository _wikiRepo;
    private readonly WikiHistoryRepository _historyRepo;
    public WikiReviewService(WikiReviewRepository repo, WikiRepository wikiRepo, WikiHistoryRepository historyRepo)
    { _repo = repo; _wikiRepo = wikiRepo; _historyRepo = historyRepo; }

    public async Task<WikiReviewResponse> CreateAsync(WikiReviewCreateRequest req)
    {
        var wiki = await _wikiRepo.GetByIdAsync(req.WikiId) ?? throw new ArgumentException("Wiki不存在");
        var r = new WikiReview { WikiId = req.WikiId, Texts = req.Texts, Tags = req.Tags, Version = wiki.Version + 0.01, UpdateTime = DateTime.Now, UpdateUser = req.UpdateUser, WikiStates = 0 };
        r.WikireviewId = await _repo.InsertAsync(r);
        return ToResponse(r);
    }

    public async Task<PageResponse<WikiReviewListResponse>> PageQueryAsync(WikiReviewPageQueryRequest req)
    {
        var (items, total) = await _repo.PageQueryAsync(req);
        return new PageResponse<WikiReviewListResponse>
        {
            Total = total, Page = req.Page, PageSize = req.PageSize,
            Data = items.Select(r => new WikiReviewListResponse { WikireviewId = r.WikireviewId, WikiId = r.WikiId, WikiStates = r.WikiStates, UpdateTime = r.UpdateTime, UpdateUser = r.UpdateUser }).ToList()
        };
    }

    public async Task<WikiReviewResponse?> UpdateStateAsync(WikiReviewUpdateRequest req)
    {
        var review = await _repo.GetByIdAsync(req.WikireviewId);
        if (review == null) return null;
        await _repo.UpdateStateAsync(req.WikireviewId, req.WikiStates);
        // If approved, update wiki
        if (req.WikiStates == 1)
        {
            var wiki = await _wikiRepo.GetByIdAsync(review.WikiId);
            if (wiki != null)
            {
                await _historyRepo.InsertAsync(new WikiHistory { WikiId = wiki.WikiId, KeyName = wiki.KeyName, Texts = wiki.Texts, Tags = wiki.Tags, Version = wiki.Version, CreateTime = wiki.CreateTime, CreateUser = wiki.CreateUser, UpdateTime = wiki.UpdateTime, UpdateUser = wiki.UpdateUser, BackupTime = DateTime.Now });
                await _wikiRepo.UpdateAsync(wiki.WikiId, review.Texts, review.Tags, review.UpdateUser);
            }
        }
        review.WikiStates = req.WikiStates;
        return ToResponse(review);
    }

    private static WikiReviewResponse ToResponse(WikiReview r) => new()
    {
        WikireviewId = r.WikireviewId, WikiId = r.WikiId, Texts = r.Texts, Tags = r.Tags,
        Version = r.Version, UpdateTime = r.UpdateTime, UpdateUser = r.UpdateUser, WikiStates = r.WikiStates
    };
}
